src/elfpy/markets.py

```diff
@@ -363,37 +472,185 @@ def _close_long(self, trade_details):
         """
         trade_results = self.pricing_model.calc_out_given_in(
             trade_details.trade_amount,
-            trade_details.share_reserves,
-            trade_details.bond_reserves,
+            trade_details.market.share_reserves,
+            trade_details.market.bond_reserves,
             trade_details.token_out,
-            trade_details.fee_percent,
+            trade_details.market.fee_percent,
             trade_details.stretched_time_remaining,
-            trade_details.init_share_price,
-            trade_details.share_price,
+            trade_details.market.init_share_price,
+            trade_details.market.share_price,
         )
         (
             without_fee_or_slippage,
             output_with_fee,
             output_without_fee,
             fee,
         ) = trade_results
-        market_deltas = {
-            "d_base_asset": -output_with_fee,
-            "d_token_asset": trade_details.trade_amount,
-            "d_base_asset_slippage": abs(without_fee_or_slippage - output_without_fee),
-            "d_token_asset_slippage": 0,
-            "d_base_asset_fee": fee,
-            "d_token_asset_fee": 0,
-            "d_base_asset_orders": 1,
-            "d_token_asset_orders": 0,
-            "d_base_asset_volume": output_with_fee,
-            "d_token_asset_volume": 0,
-        }
-        wallet_deltas = {
-            "base_in_wallet": output_with_fee,
-            "base_in_protocol": [trade_details.mint_time, 0],
-            "token_in_wallet": [trade_details.mint_time, -1 * trade_details.trade_amount],
-            "token_in_protocol": [trade_details.mint_time, 0],
-            "fee": [trade_details.mint_time, fee],
-        }
-        return market_deltas, wallet_deltas
+        market_deltas = MarketDeltas(
+            d_base_asset=-output_with_fee,
+            d_token_asset=+trade_details.trade_amount,
+            d_share_buffer=-trade_details.trade_amount / trade_details.market.share_price,
+            d_share_fee=+fee / trade_details.market.share_price,
+            d_share_fee_history={trade_details.mint_time: fee / trade_details.market.share_price},
+            d_base_asset_slippage=+abs(without_fee_or_slippage - output_without_fee),
+            d_base_asset_orders=+1,
+            d_base_asset_volume=+output_with_fee,
+        )
+        agent_deltas = AgentWallet(
+            base_in_wallet=+output_with_fee,
+            token_in_wallet={trade_details.mint_time: -trade_details.trade_amount},
+            fees_paid=+fee,
+        )
+        return market_deltas, agent_deltas
+
+    def _add_liquidity(self, trade_details):
+        """
+        Computes new deltas for bond & share reserves after liquidity is added
+        """
+        lp_out, d_base_reserves, d_token_reserves = self.pricing_model.calc_lp_out_given_tokens_in(
+            base_asset_in=trade_details.trade_amount,
+            share_reserves=trade_details.share_reserves,
```

Comment:
Should these be `trade_details.market.share_reserves`, etc? or `self.share_reserves`, as mentioned in the other comment?